sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusRetryPolicy.cs

```diff
@@ -102,15 +102,108 @@ public abstract class ServiceBusRetryPolicy
+            T1 t1,
+            T2 t2,
+            T3 t3,
+            T4 t4,
+            T5 t5,
+            TransportConnectionScope scope,
+            CancellationToken cancellationToken) => RunOperation<T1, T2, T3, T4, T5, Func<T1, T2, T3, T4, T5, TimeSpan, CancellationToken, Task>, VoidResult>(async (t11, t22, t33, t44, t55, op, timeout, token) =>
+        {
+            await op(t11, t22, t33, t44, t55, timeout, token).ConfigureAwait(false);
+            return default;
+        }, t1, t2, t3, t4, t5, operation, scope, cancellationToken);
+
+        internal Task<TResult> RunOperation<T1, TResult>(
+            Func<T1, TimeSpan, CancellationToken, Task<TResult>> operation,
+            T1 t1,
+            TransportConnectionScope scope,
+            CancellationToken cancellationToken) => RunOperation((t11, op, timeout, token) => op(t11, timeout, token),
+            t1, operation, scope, cancellationToken);
+
+        internal Task<TResult> RunOperation<T1, T2, TResult>(
+            Func<T1, T2, TimeSpan, CancellationToken, Task<TResult>> operation,
+            T1 t1,
+            T2 t2,
+            TransportConnectionScope scope,
+            CancellationToken cancellationToken) => RunOperation((t11, t22, op, timeout, token) => op(t11, t22, timeout, token),
+            t1, t2, operation, scope, cancellationToken);
+
+        internal Task<TResult> RunOperation<T1, T2, T3, TResult>(
+            Func<T1, T2, T3, TimeSpan, CancellationToken, Task<TResult>> operation,
+            T1 t1,
+            T2 t2,
+            T3 t3,
+            TransportConnectionScope scope,
+            CancellationToken cancellationToken) => RunOperation((t11, t22, t33, op, timeout, token) => op(t11, t22, t33, timeout, token),
+        t1, t2, t3, operation, scope, cancellationToken);
+
+        internal Task<TResult> RunOperation<T1, T2, T3, T4, TResult>(
+            Func<T1, T2, T3, T4, TimeSpan, CancellationToken, Task<TResult>> operation,
+            T1 t1,
+            T2 t2,
+            T3 t3,
+            T4 t4,
+            TransportConnectionScope scope,
+            CancellationToken cancellationToken)  => RunOperation((t11, t22, t33, t44, op, timeout, token) => op(t11, t22, t33, t44, timeout, token),
+            t1, t2, t3, t4, operation, scope, cancellationToken);
+
+        internal Task<TResult> RunOperation<T1, T2, T3, T4, T5, TResult>(
+            Func<T1, T2, T3, T4, T5, TimeSpan, CancellationToken, Task<TResult>> operation,
+            T1 t1,
+            T2 t2,
+            T3 t3,
+            T4 t4,
+            T5 t5,
+            TransportConnectionScope scope,
+            CancellationToken cancellationToken)  => RunOperation((t11, t22, t33, t44, t55, op, timeout, token) => op(t11, t22, t33, t44, t55, timeout, token),
+            t1, t2, t3, t4, t5, operation, scope, cancellationToken);
+
+        internal async Task<TResult> RunOperation<T1, T2, T3, T4, T5, T6, TResult>(
```

Comment:
I confess this is a bit mind-boggling... Also, maybe you have some preferences in how to name the inner params

Reply:
Would having a single overload and shuttling the parameters via the valuetyple work?

Reply:
I think this is much more elegant and potentially JIT friendly (*) and avoids the value type creation. In addition, having to make sure the value tuple fields have proper names to avoid the `Item{N}` and then having to do `valueTuple.Name` also doesn't really improve readability on the caller side. 

(*) I confess I haven't fully thought this statement through

Reply:
You can deconstruct directly into locals. Doesn't require naming fields:

``` C#
Run(static state => {
var (sender, msgs, timeout, token) = state;
...

}, sender, msgs, timeout, token)
```

>I think this is much more elegant and potentially JIT friendly (*) and avoids the value type creation.

I doubt the tuple creation would show in any meaningful way considering how expensive the new async operation is.

Reply:
Fair enough, yet here we don't require the value tuple at all. I think that is an OK trade off for a bit of complexity in the retry policy. I'll let other's chime in first.

Reply:
I could potentially try to benchmark both approaches to see how the call stack depth in the current approach impacts the perf vs value tuple creations and the deconstruction

Reply:
I'm less worried about the stack depth and more about the complexity and future support costs.

Reply:
Changed to value tuple

Reply:
Just want to make sure I understand the plan here - are we going to be able to avoid all the overloads?

Reply:
Nevermind, I see this is already updated.